Sources/SwiftParser/StringLiterals.swift

```diff
@@ -595,10 +595,105 @@ extension Parser {
       )
     }
   }
+
+  mutating func parseSimpleString() -> RawSimpleStringLiteralExprSyntax {
+    let openDelimiter = self.consume(if: .rawStringPoundDelimiter)
+    let (unexpectedBeforeOpenQuote, openQuote) = self.expect(anyIn: SimpleStringLiteralExprSyntax.OpeningQuoteOptions.self, default: .stringQuote)
+
+    /// Parse segments.
+    var segments: [RawStringSegmentSyntax] = []
+    var loopProgress = LoopProgressCondition()
+    while hasProgressed(&loopProgress) {
+      // If we encounter a token with leading trivia, we're no longer in the
+      // string literal.
+      guard currentToken.leadingTriviaText.isEmpty else { break }
+
+      if let stringSegment = self.consume(if: .stringSegment, TokenSpec(.identifier, remapping: .stringSegment)) {
+        var unexpectedAfterContent: RawUnexpectedNodesSyntax?
+
+        if let (backslash, leftParen) = self.consume(if: .backslash, followedBy: .leftParen) {
+          var unexpectedTokens: [RawSyntax] = [RawSyntax(backslash), RawSyntax(leftParen)]
+
+          let (unexpectedBeforeRightParen, rightParen) = self.expect(TokenSpec(.rightParen, allowAtStartOfLine: false))
+          unexpectedTokens += unexpectedBeforeRightParen?.elements ?? []
+          unexpectedTokens.append(RawSyntax(rightParen))
+
+          unexpectedAfterContent = RawUnexpectedNodesSyntax(
+            unexpectedTokens,
+            arena: self.arena
+          )
+        }
+
+        segments.append(RawStringSegmentSyntax(content: stringSegment, unexpectedAfterContent, arena: self.arena))
+      } else {
+        break
+      }
+    }
+
+    let (unexpectedBetweenSegmentAndCloseQuote, closeQuote) = self.expect(
+      anyIn: SimpleStringLiteralExprSyntax.ClosingQuoteOptions.self,
+      default: openQuote.closeTokenKind
+    )
+    let closeDelimiter = self.consume(if: .rawStringPoundDelimiter)
+
+    if openQuote.tokenKind == .multilineStringQuote, !openQuote.isMissing, !closeQuote.isMissing {
+      let postProcessed = postProcessMultilineStringLiteral(
+        rawStringDelimitersToken: openDelimiter,
+        openQuote: openQuote,
+        segments: segments.compactMap { RawStringLiteralSegmentListSyntax.Element.stringSegment($0) },
+        closeQuote: closeQuote
+      )
+
+      return RawSimpleStringLiteralExprSyntax(
+        RawUnexpectedNodesSyntax(
+          combining: unexpectedBeforeOpenQuote,
+          postProcessed.unexpectedBeforeOpeningQuote,
+          openDelimiter,
```

Comment:
I would revers this order because `unexpectedBeforeOpeningQuote` is directly before the opening quote and thus after the delimiter.

```suggestion
          combining: openDelimiter,
          unexpectedBeforeOpenQuote,
          postProcessed.unexpectedBeforeOpeningQuote,

```